src/test/java/org/gridsuite/securityanalysis/server/util/MatcherJson.java

```diff
@@ -0,0 +1,114 @@
+/*
+  Copyright (c) 2021, RTE (http://www.rte-france.com)
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.securityanalysis.server.util;
+
+import java.util.Collection;
+
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+
+/**
+ * @author Slimane Amar <slimane.amar at rte-france.com>
+ * @author Laurent Garnier <laurent.garnier at rte-france.com>
+ */
+public class MatcherJson<T> extends TypeSafeMatcher<T> {
```

Comment:
Are you sure you don't have a simpler solution (lombok annotation toString) ?
Because it cannot easily be maintained
When there is too much override that means there is a problem ?

Reply:
Just pushed the simpler version with mismatch description on one line, with the mapper used for the comparison